File: backend/tools/desktop_control/desktop_wait.py
"""Short pause during desktop automation. Runs on the Akira server host."""
from backend.tools.desktop_control._desktop_control_core import call_desktop_tool

TOOL_DEF = {
    "name": "desktop_wait",
    "description": (
        "Sleep up to 30 seconds on the Akira server between UI steps (animations, loads). "
        "Does not move the mouse; use sparingly. Enable only when you trust automation on this PC."
    ),
    "input_schema": {
        "type": "object",
        "properties": {
            "seconds": {
                "type": "number",
                "description": "Duration to wait (0–30 seconds).",
            },
        },
        "required": ["seconds"],
    },
    "default_enabled": False,
    "timeout_seconds": 300,
}


def call_tool(tool_input: dict, context=None):
    return call_desktop_tool("wait", tool_input, context)

File: akira-desktop/electron/tools/desktop-automation/desktop-wait.js
/**
 * Desktop Wait Tool
 * desktop_wait - Wait/sleep for a specified duration (0-30 seconds)
 */

const definitions = [
  {
    name: 'desktop_wait',
    description: 'Wait/sleep for a specified duration (0-30 seconds). Use between UI actions.',
    input_schema: {
      type: 'object',
      properties: {
        seconds: {
          type: 'number',
          description: 'Duration to wait (0-30 seconds)',
        },
      },
      required: ['seconds'],
    },
  },
];

const handlers = {
  async desktop_wait(input) {
    const seconds = input.seconds;

    if (seconds == null || seconds < 0 || seconds > 30) {
      return { success: false, error: 'seconds must be between 0 and 30' };
    }

    await new Promise(resolve => setTimeout(resolve, seconds * 1000));
    return { waited_seconds: seconds };
  },
};

module.exports = { definitions, handlers };
